src/utils/compose_ts.ts

```diff
@@ -696,7 +697,7 @@ export async function getServiceDirsFromComposition(
 			let dir =
 				(typeof service.build === 'string'
 					? service.build
-					: service.build?.context) || '.';
+					: service.build?.context) ?? '.';
```

Comment:
Please confirm whether `''` is possible and whether that's okay. Also (not part of your change), odd that `service.build` is type `never`. The typing does not indicate that it can ever be a string 🤔

Reply:
A user could explicitly set context to '' which then defaults to cwd in native Docker compose, so it appears an empty string is valid.

Regarding the typing being `never`, that's actually because balena-compose types BuildConfig wrongly. I'm fixing that with my next PR to integrate a newer balena-compose into CLI